impl/src/main/kotlin/nl/nuts/consent/bridge/rpc/CordaRPClientFactory.kt

```diff
@@ -117,44 +133,58 @@ class CordaRPClientWrapper : AutoCloseable {
         val nodeAddress = NetworkHostAndPort(consentBridgeRPCProperties.host, consentBridgeRPCProperties.port)
 
         do {
-            connection = try {
-                logger.info("Connecting to: $nodeAddress")
-                val client = CordaRPCClient(
-                        nodeAddress,
-                        object : CordaRPCClientConfiguration() {
-                            override val connectionMaxRetryInterval = retryInterval
-                        }
-                )
-                val unvalidatedConnection = client.start(
-                        username = consentBridgeRPCProperties.user,
-                        password = consentBridgeRPCProperties.password,
-                        gracefulReconnect = GracefulReconnect(
-                                maxAttempts = -1,
-                                onDisconnect = { logger.error("disconnected from RPC") },
-                                onReconnect = { logger.info("RPC connection re-established") }
-                        )
-                )
-
-                // Check connection is truly operational before returning it.
-                val nodeInfo = unvalidatedConnection.proxy.nodeInfo()
-                require(nodeInfo.legalIdentitiesAndCerts.isNotEmpty())
-                unvalidatedConnection
-            } catch(secEx: ActiveMQSecurityException) {
-                // Incorrect credentials, log and rethrow
-                logger.error(secEx.message)
-                throw secEx
-            } catch(ex: RPCException) {
-                // Deliberately not logging full stack trace as it will be full of internal stacktraces.
-                logger.error("Exception upon establishing connection: " + ex.message)
-                retryCount--
-                if (retryCount == 0) {
-                    shutdown = true
+            try {
+                // first try socket, because RPCCLient will hang indefinitely
+                val clientSocket = Socket()
+                clientSocket.connect(InetSocketAddress(consentBridgeRPCProperties.host, consentBridgeRPCProperties.port), retryInterval.toMillis().toInt())
+                clientSocket.close()
+
+                connection = try {
+                    logger.info("Connecting to: $nodeAddress")
+
+                    val gracefulReconnect = GracefulReconnect(
+                            maxAttempts = consentBridgeRPCProperties.retryCount,
+                            onDisconnect = { logger.error("disconnected from RPC") },
```

Comment:
Not sure if a disconnection should be an ERROR level message. If it can gracefully recover from it without human intervention (because it's reconnecting), maybe it should be a WARN.

Reply:
I'd rather have it popup as an error for now to get a feeling on how often it occurs and how harmful it is.